dht.go

```diff
@@ -685,13 +707,22 @@ func (dht *IpfsDHT) FindLocal(id peer.ID) peer.AddrInfo {
 
 // nearestPeersToQuery returns the routing tables closest peers.
 func (dht *IpfsDHT) nearestPeersToQuery(pmes *pb.Message, count int) []peer.ID {
+	key := pmes.GetKey()
+
 	if pmes.GetType() == pb.Message_GET_PROVIDERS {
 		// for GET_PROVIDERS messages, the message key is the hashed multihash, so don't hash it again
-		closer := dht.routingTable.NearestPeers(kb.ID(string(pmes.GetKey())), count)
-		return closer
+		if len(key) < 32 {
```

Comment:
note to self: since prefixes are now in bits, the key byte length can be 32 but it can still be a prefix lookup. maybe add a specific `lookupKey` type that contains the key and the prefix length?